<launch>
<arg name="robot_id"/>
<arg name="base_station"/>
<arg name="robot_type"/>
<arg name="relay_robots" default="[]"/>
<node name="BaseAlign" pkg="tf" type="static_transform_publisher" args="0 0 0 3.14 0 0  base_link  laser 10"/>

<group ns="robot_$(arg robot_id)">
    <rosparam file="ros_robot.yaml"/>

    <include file="$(find gvgexploration)/launch/robot.launch">
        <arg name="robot_name" value="$(arg robot_id)" />
        <arg name="robot_type" value="$(arg robot_type)" />
        <arg name="base_stations" value="$(arg base_station)" />
    </include>
    <node pkg="nav2d_operator" type="operator" name="operator" output="screen">
        <rosparam file="operator_robot.yaml"/>
        <rosparam file="costmap1_robot.yaml" ns="local_map" />
        <param name="robot_frame" value="base_link"/>
        <param name="odometry_frame" value="odom"/>
    </node>
    <node pkg="nav2d_navigator" type="navigator" name="navigator">
        <rosparam file="navigator_robot.yaml"/>
    </node>
    <node name="GetMap" pkg="nav2d_navigator" type="get_map_client" />
    <node name="SetGoal" pkg="nav2d_navigator" type="set_goal_client" />

    <node pkg="rosbot_webui" type="serial_bridge.sh" name="serial_bridge" output="screen"/>

    <include file="$(find rplidar_ros)/launch/rplidar.launch"/>

    <node pkg="nav2d_karto" name="Mapper" type="mapper">
        <rosparam command="load" file="mapper_robot.yaml" subst_value="true"/>
    </node>

   <node name="master_discovery" pkg="master_discovery_fkie" type="master_discovery" >
        <param name="_mcast_group" value="224.0.0.1"/>
    </node>
    <node name="master_sync" pkg="master_sync_fkie" type="master_sync" output="screen">
        <rosparam param="ignore_topics">['/rosout','/rosout_agg','/shared_data_size','/shutdown','/tf','/tf_static']</rosparam>
        <rosparam param="ignore_nodes"> ['/*msgs_conversion*','/*rosbot_ekf*','/*rplidarNode*','/*serial_node*','/*Mapper*','/*GetMap*','/*OdomAlign*','/BaseAlign','/MapAlign','/*SetGoal*','/*navigator*','/*operator*'] </rosparam>
    </node>

</group>
    <node pkg="rosbag" type="record" name="gvg_rosbag" output="screen" args="--all"/>

</launch>
